src/quick_sort.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import timeit
+
+
+BEST_CASE = [x for x in range(0, 1000)]
+WORST_CASE = BEST_CASE[::-1]
+
+
+def quick_sort(a_list):
+    """Use quick sort to sort the list."""
+    left = []
+    right = []
+    if len(a_list) <= 1:
+        return a_list
+    if len(a_list) == 2:
+        if a_list[0] < a_list[1]:
+            left.append(a_list[0])
+            right.append(a_list[1])
+        else:
+            left.append(a_list[1])
+            right.append(a_list[0])
+    else:
+        hi = a_list[-1]
+        low = a_list[0]
+        midpoint = a_list[len(a_list)//2]
+        pivot = sorted([low, hi, midpoint])[1]
+        for item in a_list:
+            if item < pivot:
```

Comment:
This will work in a lot of cases, but there is another scenario which you are not considering, and for which you need to test. Currently your algorithm is getting stuck in an infinite loop.

Make sure you include some test cases with duplicate items in your list to be sorted.